test/modules/layers/test_conv.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import unittest
+from itertools import product
+
+import torch
+from test.test_utils import assert_expected
+from torchmultimodal.modules.layers.conv import (
+    calculate_same_padding,
+    calculate_transpose_padding,
+    SamePadConv3d,
+    SamePadConvTranspose3d,
+)
+
+
+class TestSamePadConv3d(unittest.TestCase):
+    """
+    Test the SamePadConv3d class and associated helpers
+    """
+
+    def setUp(self):
+        inputs = [torch.ones(1, 1, 8, 8, 8), torch.ones(1, 1, 7, 7, 7)]
+        kernels = [(4, 4, 4), (3, 3, 3)]
+        strides = [(2, 2, 2), (3, 3, 3)]
+        self.test_cases = list(product(*[inputs, kernels, strides]))
```

Comment:
so I do want to test all these test cases, as they are each important (even vs odd input size, even vs odd kernel, even vs odd stride). I could list out all the combinations manually since there should be only 8 cases, I was initially thinking it wouldn't be much more readable seeing multiple long lists. let me make the change and we can reevaluate

Reply:
Sounds good. I actually don't have an especially strong stance here, if listing them all out leads to a ton of boilerplate code it's also fine to leave as it is.